### ISSUE TITLE
RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by DAmNRelentless** [simonhaenisch/prettier-plugin-organize-imports#13]
Hello there!

I recently switched from JavaScript to TypeScript. I always liked Prettier to keep a consistent code style.
I found your extension very promising but I ran into an issue when using it.

I have several plugins and packages on my TypeScript project but after trying for a long time, I was able to reproduce it and it definitely is caused by your plugin since it works without it.

When I save a TS file in my larger TS project, it won't be formatted at all. It seems like prettier crashes completely.
I have a little exclamation mark in the task bar next to the prettier label and if I click on it, the following message appears:
```
["ERROR" - 7:38:37 PM] Error formatting document.
RangeError: Maximum call stack size exceeded
```
The whole log can is here: 
[log.log](https://github.com/simonhaenisch/prettier-plugin-organize-imports/files/5740337/log.log)

Although it says "Formatting completed" at the end, nothing happens.
And this doesn't happen without this plugin.

Versions:
organize imports: v1.1.1
prettier: v2.2.1
VSCode: 1.52.1
OS: Windows 10 64 bit 2004 but my TS project is living in a WSL2 environment running on Ubuntu 20.04

**Comment by simonhaenisch** [comment]
Just looking at the call stack, it looks like the code that causes this is part of the TypeScript lib (`processCommentPragmas`). All this plugin does is call `organizeImports` of the TypeScript language service, so I wonder whether you run into the same issue when you use the "Organize Imports" action via the command palette (<kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>P</kbd>)?

Is it a very large file that you're trying to format? You wouldn't be able to provide the file for debugging purposes, would you? Otherwise I'm sorry but I think this is out of my hands as the issue is most likely upstream (https://github.com/microsoft/TypeScript/blob/master/src/services/organizeImports.ts).

**Comment by DAmNRelentless** [comment]
So sorry! Didn't see you replied to the issue. I usually check my notifications every day multiple times.
Must have skipped that.

I tried using the `organizeImports` option from the command palette and it works fine.
I am not trying to format a large file. I can even create a new empty file and I will get the same result.

**Comment by simonhaenisch** [comment]
Hm ok, however I'm sorry but I'm not able to reproduce this. I don't have a Windows machine around.

I've added cross-platform matrix testing to the CI now to make sure the plugin code works on all platforms. It does: https://github.com/simonhaenisch/prettier-plugin-organize-imports/actions/runs/468671986.

The only thing I can think of is that it's an issue with the TypeScript version in your project. From looking at your log file I can tell that VS Code not using the built-in TypeScript version but the one that's locally installed into your project. Which one is that?

**Comment by DAmNRelentless** [comment]
I am using Windows 10 Pro but my whole development environment is sitting in the WSL 2 environment which is running Ubuntu 20.02.
I am running TypeScript 4.1.3 in my project.
Node is running on 15.5.0.

**Comment by simonhaenisch** [comment]
Ok I'm using similar versions (same prettier, same plugin, same typescript, and Node 15.3), and everything works fine for me (however I'm on macOS).

What you can try to do is invoke `prettier` from the command line in your project:

```
npx prettier --write path/to/some/file.ts
```

If that runs into the same issue, then at least we know it's not related to VS Code. In that case, can you also try to create a clean project and see whether it reproduces?

```
mkdir organize-imports-test
cd organize-imports-test

npm init -y
npm install --save-dev prettier prettier-plugin-organize-imports

touch file.ts

npx prettier --write file.ts
```

**Comment by DAmNRelentless** [comment]
It looks like I am not able to reproduce this behavior anymore.
I reinstalled the plugin and tried running prettier from the command line like you said and it worked.

After that, I tried auto-formatting by saving again but it worked without any error. Really odd.
I reinstalled my OS in the last few days because I had problems with the update and that seems to have fixed it for now.

I will reopen the issue if I am able to reproduce this again. Thanks for your help!
